frontend/src/features/apartment/ApartmentSalesPage.tsx

```diff
@@ -38,66 +39,134 @@ const MaximumPriceModalError = ({error, setIsModalVisible}) => {
+                setMaxPrices({
+                    maximumPrice: apartment.prices.maximum_prices.confirmed?.maximum_price,
+                    maxPricePerSquare: maxPriceCalculation
+                        ? maxPriceCalculation.calculations.construction_price_index.calculation_variables
+                              .debt_free_price_m2
+                        : maxPriceData?.calculations.construction_price_index.calculation_variables.debt_free_price_m2,
+                    debtFreePurchasePrice: maxPriceCalculation
+                        ? maxPriceCalculation.calculations.construction_price_index.calculation_variables
+                              .debt_free_price
+                        : maxPriceData?.calculations.construction_price_index.calculation_variables.debt_free_price,
+                    index: maxPriceData ? maxPriceData.index : maxPriceCalculation?.index,
+                });
+                setIsModalVisible(true);
+            });
+        } else
+            hitasToast(
+                <>
+                    Enimmäishintojen laskemiseen tarvitaan <span>kauppakirjan päivämäärä</span> sekä{" "}
+                    <span>yhtiön lainaosuus</span>!
+                </>,
+                "error"
+            );
     };
+    const handleSaveButton = () => {
+        if (formOwnershipsList.length < 1) {
+            hitasToast("Asunnolla täytyy olla ainakin yksi omistaja!", "error");
+            setHasNoOwnershipsError(true);
+        } else {
+            const saleData = {
+                ...formData,
+                ownerships: formOwnershipsList,
+                exclude_from_statistics: isExcludedFromStats,
+            };
+            saveSale({
+                data: saleData,
+                apartmentId: apartment.id,
+                housingCompanyId: apartment.links.housing_company.id,
+            });
+        }
+    };
+
+    // Handle saving flow
+    useEffect(() => {
+        if (!isLoading && !error && data && data.id) {
+            hitasToast("Kauppa tallennettu onnistuneesti!");
+            navigate(`/housing-companies/${apartment.links.housing_company.id}/apartments/${apartment.id}`);
+        } else if (error) {
+            hitasToast("Kaupan tallentaminen epäonnistui.", "error");
+        }
+    }, [isLoading, error, data, navigate, apartment.links.housing_company.id, apartment.id]);
+    // Handle alert for changed loan value
+    useEffect(() => {
+        if (
+            Number(formData.apartment_share_of_housing_company_loans) !==
+                maxPriceCalculation?.calculations[maxPriceCalculation.index].calculation_variables
+                    .apartment_share_of_housing_company_loans &&
+            maxPriceCalculation
+        )
+            setIsLoanValueChanged(true);
+        else setIsLoanValueChanged(false);
+    }, [
+        formData.apartment_share_of_housing_company_loans,
+        maxPriceCalculation,
+        setIsLoanValueChanged,
+        isLoanValueChanged,
+    ]);
```

Comment:
`isLoanValueChanged` does not need its own state, as the value is changed as a reaction to some other values being changed, and there are no things that need to react to `isLoanValueChanged` itself being changed.

So, instead of having a state for `isLoanValueChanged`, we can simply store the value in a const.
```ts
    const isLoanValueChanged =
        Number(formData.apartment_share_of_housing_company_loans) !==
            maxPriceCalculation?.calculations[maxPriceCalculation.index].calculation_variables
                .apartment_share_of_housing_company_loans && maxPriceCalculation;
```

Reply:
Well spotted! Should have maybe gotten that myself from the fact that isPriceTooHigh works as a const, too! 😅